File: php/src/Normal.php
<?php

namespace GildedRose;

class Normal implements Product
{
    public function updateQuality(object $item): void
    {
        if ($item->quality > 0) {
            $item->quality = $item->quality - 1;
        }

        $item->sellIn = $item->sellIn - 1;

        if ($item->sellIn < 0) {
            if ($item->quality > 0) {
                $item->quality = $item->quality - 1;
            }
        }
    }
}

File: go/gildedrose/strategy/normal.go
package strategy

import "github.com/emilybache/gildedrose-refactoring-kata/gildedrose/item"

type Normal struct {
	*item.Item
}

func (item *Normal) UpdateSellIn() {
	item.Item.SellIn -= 1
}

func (item *Normal) UpdateQuality() {
	switch {
	case item.Item.SellIn <= 0:
		item.Item.Quality -= 2
	default:
		item.Item.Quality -= 1
	}
	if item.Item.Quality < 0 {
		item.Item.Quality = 0
	}
}
